extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/export/PrometheusRecorder.java

```diff
@@ -24,7 +23,7 @@ public PrometheusHandler getHandler() {
         return new Function<Router, Route>() {
             @Override
             public Route apply(Router router) {
-                return router.route(path).produces(TextFormat.CONTENT_TYPE_004);
+                return router.route(path).order(1).produces("text/plain");
```

Comment:
I might have missed this somewhere, but should the content type be changed? I thought "004" was the right one for Prometheus?

Reply:
Jan and I talked about this one. Due to precedence order, it's safer to use text plain... if the suffix (for type 004 is there), it gets weird with JSON being on the same endpoint. 

Can two types be produced in different orders? (text/plain at order(1) AND 004)?

Reply:
This is only for handler matching purposes, this is not what will be sent to the client. The actual content type is set in the handler: https://github.com/quarkusio/quarkus/blob/1.8.1.Final/extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/export/handlers/PrometheusHandler.java#L34

The thing is that there are now potentially two distinct handlers living on the same `/metrics` endpoint. Vert.x chooses the handler based on this `produces` value, and if we left this value as `text/plain; version=0.0.4; charset=utf-8`, then it would only be matched if the client sent an `Accept` header with this whole value. So if the client requested `Accept: text/plain` there would be no matched handler to handle the request.

Reply:
Aha, now I remember that.

Thanks for the explanation!